crates/adapters/dydx/src/grpc/builder.rs

```diff
@@ -0,0 +1,163 @@
+// -------------------------------------------------------------------------------------------------
+//  Copyright (C) 2015-2025 Nautech Systems Pty Ltd. All rights reserved.
+//  https://nautechsystems.io
+//
+//  Licensed under the GNU Lesser General Public License Version 3.0 (the "License");
+//  You may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at https://www.gnu.org/licenses/lgpl-3.0.en.html
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------------------------------
+
+//! Transaction builder for dYdX v4 protocol.
+//!
+//! This module provides utilities for building and signing Cosmos SDK transactions
+//! for the dYdX v4 protocol.
+
+use std::fmt::Debug;
+
+use cosmrs::{
+    Any, Coin,
+    tx::{self, Fee, SignDoc, SignerInfo},
+};
+use rust_decimal::{Decimal, prelude::ToPrimitive};
+
+use super::{types::ChainId, wallet::Account};
+
+/// Gas adjustment value to avoid rejected transactions caused by gas underestimation.
+const GAS_MULTIPLIER: f64 = 1.8;
```

Comment:
How did we determine this?

Reply:
from [here](https://github.com/dydxprotocol/v4-clients/blob/main/v4-client-rs/client/src/node/fee.rs#L11)